### PR TITLE
Improve generate-* targets in Makefile

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
This PR attempts to cleanup the Makefile targets specifically around the generation. It reduces build time and the amount of times we call `dep-ensure` and `gazelle`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #945  

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,12 +81,18 @@ lint-full: dep-ensure ## Run slower linters to detect possible issues
 	bazel run //:lint-full $(BAZEL_ARGS)
 
 .PHONY: generate
-generate: clientset dep-ensure ## Generate code
-	go generate ./pkg/... ./cmd/...
+generate: dep-ensure ## Generate code
```
Comment (ncdc) [inline Makefile]: Why do we need a dependency on `dep-ensure` here?

File: hack/verify_clientset.sh [hunk 2]
```diff
@@ -35,7 +35,7 @@ cleanup
 mkdir -p "${TMP_DIFFROOT}"
 cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}"
 
-make clientset
+make generate
```
Comment (ncdc) [inline hack/verify_clientset.sh]: `generate-clientset` instead?

**Comment by ncdc** [comment]
I didn't put this on individual lines, but in general, to me, `make generate` should be something you consciously decide to do, and it's unlikely it will happen frequently. How would you feel about removing `generate` as a dependency from targets such as `manager`, `clusterctl`, and so on?

**Comment by vincepri** [comment]
@ncdc Done

**Comment by vincepri** [comment]
This should be ready to go, I'll rebase the other open PR on it when merged

**Pull request merged**